ydb/core/tx/scheme_board/populator.cpp

```diff
@@ -705,6 +707,37 @@ class TPopulator: public TMonitorableActor<TPopulator> {
         }
     }
 
+    bool CheckQuorum(TVector<ui32> &replicaRepliesCounter, TActorId foundReplica) {
+        ui32 quorumCnt = 0;
+        for (const auto& ringGroup : GroupInfo->RingGroups) {
+            if (!ringGroup.WriteOnly) {
+                ++quorumCnt;
+            }
+        }
+        TVector<bool> quorum(GroupInfo->RingGroups.size());
+
+        for (ui32 ringGroupIndex : xrange(GroupInfo->RingGroups.size())) {
+            const auto& ringGroup = GroupInfo->RingGroups[ringGroupIndex];
+            if (ringGroup.WriteOnly) {
+                continue;
+            }
+            for (const auto& ring : ringGroup.Rings) {
+                for (const auto& replica : ring.Replicas) {
+                    if (replica == foundReplica 
+                        && ++replicaRepliesCounter[ringGroupIndex] > (ringGroup.NToSelect / 2) 
+                        && !quorum[ringGroupIndex]) {
```

Comment:
```suggestion
                        && !quorum[ringGroupIndex])
                    {
```
Иначе сливается условие и тело.